src/JBGovernanceNFT.sol

```diff
@@ -1,159 +1,120 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.17;
 
-import {JBGovernanceNFTStake} from "./structs/JBGovernanceNFTStake.sol";
 import {JBGovernanceNFTMint} from "./structs/JBGovernanceNFTMint.sol";
 import {JBGovernanceNFTBurn} from "./structs/JBGovernanceNFTBurn.sol";
 
 import "@openzeppelin/contracts/utils/Checkpoints.sol";
 import {IERC20, SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import {EIP712, ERC721, ERC721Votes} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Votes.sol";
 
+
+/**
+ * @title JB Governance NFT
+ * @dev Used for granting voting rights and staking functionality to the holder. 
+   @dev
+    Adheres to -
+    ERC721Votes: For enabling the voting and delegation mechanism.
+ */
 contract JBGovernanceNFT is ERC721Votes {
     using Checkpoints for Checkpoints.History;
     using SafeERC20 for IERC20;
 
-    event NFTStaked(uint256 _tokenId, address _stakedAt);
-    event NFTUnstaked(uint256 _tokenId);
+    event NFT_Minted_and_Staked(uint256 _tokenId, address _stakedAt);
+    event NFT_Burnt_and_Unstaked(uint256 _tokenId);
 
+    //*********************************************************************//
+    // --------------------------- custom errors ------------------------- //
+    //*********************************************************************//
     error NO_PERMISSION(uint256 _tokenId);
     error INVALID_STAKE_AMOUNT(uint256 _i, uint256 _amount);
-    error NFT_IS_STAKED(uint256 _tokenId, address _stakedAt);
 
+    /**
+     * @dev The ERC20 token that is used for staking.
+     */
     IERC20 immutable token;
+
+    /**
+     * @dev A mapping of staked token balances by user.
+     */
     mapping(address => uint256) public stakingTokenBalance;
-    mapping(uint256 => JBGovernanceNFTStake) stakes;
 
+    /**
+     * @dev The next available token ID to be minted.
+     */
     uint256 nextokenId = 1;
 
+    //*********************************************************************//
+    // ---------------------------- constructor -------------------------- //
+    //*********************************************************************//
+
+    /**
+     * @param _token The ERC20 token to use for staking.
+     */
     constructor(IERC20 _token) ERC721("", "") EIP712("", "") {
         token = _token;
     }
 
-    function mint(JBGovernanceNFTMint[] calldata _mints) external returns (uint256 _tokenId) {
+    /**
+     * @dev Mint a new NFT and stake tokens.
+     * @param _mints An array of struct containing the beneficiary and stake amount for each NFT to be minted.
+     * @return _tokenId The token ID of the newly minted NFT.
+    */
+    function mint_and_stake(JBGovernanceNFTMint[] calldata _mints) external returns (uint256 _tokenId) {
```

Comment:
nit: if we don't have any staking we could just call this `mint` (same for `burn_and_unstake`)